windowing.py

```diff
@@ -0,0 +1,198 @@
+import pickle
+import numpy as np
+from skimage.util.shape import view_as_windows
+
+
+def create_windows_from_dictmatrix(dict_matrix, rows, window_size, stride):
+    """
+    Description: takes dictionary that contains x, y, z, nanos keys and transforms
+                 this into an np array. Then creates windows from this matrix and returns
+                 the windows
+
+    Args:
+        dict_matrix -- dictionary  to create windows from
+        rows -- int (number of rows in a trace)
+        window_size -- int (# of samples within one window when creating the windows)
+        stride -- int (# of samples between the start of each window)
+
+    Return:
+        windowed_data -- np array (np array of the windowed data)
+    """
+    x = dict_matrix['x']
+    y = dict_matrix['y']
+    z = dict_matrix['z']
+    nanos = dict_matrix['nanos']
+    np_data = np.array([x, y, z, nanos], 'd')
+    windowed_data = view_as_windows(np_data, (rows, window_size), step=stride)
+    return windowed_data
+
+
+def find_initial_lift_times(lift_windows, rows):
+    """
+    Description: returns the start and end times from lift_windows np array
+
+    Args:
+        lift_windows -- np array
+        rows -- int (number of rows in a trace)
+
+    Return:
+        start -- int (start time in nanos)
+        end -- int (end time in nanos)
+    """
+    start = lift_windows[0][rows-1][0]
+    end = lift_windows[-1][-1][-1]
+    return start, end
+
+
+def find_lift_windows(windowed_data, divisor, threshold, return_lift_windows):
+    """
+    Description: takes the windowed data and calculates variance for each window.
+                 Then calculates variance threshold and returns an np array of
+                 the windows that meet the variance threshold.
+    Args:
+        windowed_data -- np array [[[[x1],[y1],[z1],[nanos1]], [[x2], [y2], [z2], [nanos2]]...]
+        divisor -- int (will use divisor to calculate variance threshold so
```

Comment:
Could you format comments to be within 80 characters per line? That helps people read them with a more standardized format.